src/java/com/zimbra/cs/taglib/tag/LoginTag.java

```diff
@@ -127,6 +135,10 @@ public void doTag() throws JspException, IOException {
             PageContext pageContext = (PageContext) jctxt;
             HttpServletRequest request = (HttpServletRequest) pageContext.getRequest();
 
+            if(request.getParameter("g-recaptcha-response")!= null && !isCaptchaValid("6LdfLE0UAAAAAC3rf4G4oepXp7H0AEEegdfeDvUA", request.getParameter("g-recaptcha-response"))){
```

Comment:
If that is supposed to be a secret key, we don't want it hard coded.

Reply:
This functionality is going to be covered in new separate ticket